### PR TITLE
Create Date Filter

### PR DESCRIPTION
**Pull request by MatthewNeufeld**
### Describe your changes

This PR accomplishes the following:

- Enables filtering out of list items outside a lower bounded start date and upper bounded end date

### Issue ticket number and link

This PR addresses: (https://github.com/CMPUT301F23T08/HouseHomey/issues/28)

### Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Necessary java docs are present
- [ ] Necessary UML diagrams are created
- [ ] Necessary tests are written

### CODE REVIEW

File: app/src/main/java/com/example/househomey/filter/ui/DateFilterFragment.java [hunk 1]
```diff
@@ -1,17 +1,104 @@
...
+    protected Date startDate;
+    protected Date endDate;
+
+    /**
+     * Constructs new DateFilterFragment
+     * @param title the title of the new date filter dialogue
+     * @param contentView the content view of the date filter dialogue
+     * @param filterCallback the callback interface for handling filter changes
+     */
     public DateFilterFragment(String title, View contentView, FilterCallback filterCallback) {
```
[the remark this answers is not in the document]
Comment (MatthewNeufeld) [reply app/src/main/java/com/example/househomey/filter/ui/DateFilterFragment.java:36]: Sounds good - glad you guys are looking into that. I'll move the InitDateField stuff to an onCreateDialo

File: app/src/main/java/com/example/househomey/filter/ui/DateFilterFragment.java [hunk 2]
```diff
@@ -1,17 +1,104 @@
...
+     * @param layoutId the ID of the layout, will correspond to layout for start or end date
+     * @return dateTextView: the
+     */
+    private TextInputEditText InitDateField(View contentView, int dateViewId, int layoutId) {
+
+        // Getting the right TextInputEditText based on dateViewId
+        TextInputEditText dateTextView = contentView.findViewById(dateViewId);
+
+        // Setup calendar requirements and constraints
+        CalendarConstraints.Builder constraintsBuilder = new CalendarConstraints.Builder();
```
[the remark this answers is not in the document]
Comment (MatthewNeufeld) [reply app/src/main/java/com/example/househomey/filter/ui/DateFilterFragment.java:56]: Yep I took that from ItemFormFragment. Could probably refactor into single util function. I'll look into it.

File: app/src/main/java/com/example/househomey/filter/ui/FilterFragment.java [hunk 3]
```diff
@@ -66,5 +72,5 @@ public void onClick(DialogInterface dialog, int which) {
      * This method should be implemented in subclasses to define filter-specific logic
      * that extract filter properites from XML like dates, make, keywords, and tags.
      */
-    public abstract void getFilterInput();
+    public abstract void getFilterInput() throws ParseException;
```
[the remark this answers is not in the document]
Comment (MatthewNeufeld) [reply app/src/main/java/com/example/househomey/filter/ui/FilterFragment.java:75]: My bad that is leftover from another implementation I tried before. Missed it when I did my code review. I'll take it out.

**Review by jdrco — COMMENTED** [review]
Hey @MatthewNeufeld everything looks great! One thing, right now there is no way to show the user what the current date filter applied is.

One way to tackle this for now is: if you apply a date filter and then try to apply a new date filter, the previous date filter should be shown/auto-filled in the start and end date inputs. That way the user knows what the current date range applied is.

**Comment by MatthewNeufeld** [comment]
Good point @jdrco I'll add that in.

**Comment by MatthewNeufeld** [comment]
> Hey @MatthewNeufeld everything looks great! One thing, right now there is no way to show the user what the current date filter applied is.
> 
> One way to tackle this for now is: if you apply a date filter and then try to apply a new date filter, the previous date filter should be shown/auto-filled in the start and end date inputs. That way the user knows what the current date range applied is.

Good point @jdrco I'll add that in.

**Comment by MatthewNeufeld** [comment]
> > Hey @MatthewNeufeld everything looks great! One thing, right now there is no way to show the user what the current date filter applied is.
> > One way to tackle this for now is: if you apply a date filter and then try to apply a new date filter, the previous date filter should be shown/auto-filled in the start and end date inputs. That way the user knows what the current date range applied is.
> 
> Good point @jdrco I'll add that in.

Gonna do this once it is merged in.

**Review by jdrco — APPROVED** [review]
LGTM

**Review by owencooke — APPROVED** [review]
Looks great after resolutions 🤖

**Pull request merged**